addons/udes_stock/tests/test_stock_warehouse.py

```diff
@@ -7,22 +7,37 @@ def setUpClass(cls):
         super(TestStockWarehouseModel, cls).setUpClass()
         cls.Warehouse = cls.env["stock.warehouse"]
 
-    def test01_get_test_picking_types(self):
+    def test_get_test_picking_types(self):
         """Test checks if the test picking types are present"""
         test_picking_types = self.warehouse.get_picking_types().filtered(
             lambda pt: "TEST" in pt.name
         )
         self.assertEqual(len(test_picking_types), 6)
 
-    def test02_get_picking_types(self):
+    def test_get_picking_types(self):
         """Test checks if the test picking types and the normal pick types are present"""
         test_picking_types = self.warehouse.get_picking_types()
         self.assertEqual(len(test_picking_types), 9)
 
-    def test03_get_default_picking_types_new_warehouse(self):
+    def test_get_default_picking_types_new_warehouse(self):
         """Create a new warehouse and look at the default picking types"""
         # Create a new company creates a new warehouse
         test_company = self.create_company("test_company")
         new_warehouse = self.Warehouse.search([("company_id", "=", test_company.id)])
         test_picking_types = new_warehouse.get_picking_types()
         self.assertEqual(len(test_picking_types), 3)
+    
+    def test_check_picking_types_prefix_after_updating_warehouse_name(self):
+        """
+        Test that making changes to the warehouse does not cause the pickingtype prefixes to reset. 
+        """ 
+        Pickingtype = self.env["stock.picking.type"]
+        
+        goods_in_prefix = Pickingtype.search([("name", "=", "Goods In")]).sequence_id.prefix
+        self.assertEqual(goods_in_prefix, "IN")
+        
+        self.warehouse.write({"code":"NEW_WAREHOUSE_CODE"})
+
+        goods_in_prefix = Pickingtype.search([("name", "=", "Goods In")]).sequence_id.prefix
+        self.assertEqual(goods_in_prefix, "IN")
```

Comment:
Can we check the full name again, so that we can see the NEW_WAREHOUSE_CODE/IN ?

Reply:
Same as comment above.